internal/cli/auth/login.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/oauth"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas/auth"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+//go:generate mockgen -destination=../../mocks/mock_login.go -package=mocks github.com/mongodb/mongocli/internal/cli/auth Authenticator
+
+type Authenticator interface {
+	RequestCode(context.Context) (*auth.DeviceCode, *atlas.Response, error)
+	PollToken(context.Context, *auth.DeviceCode) (*auth.Token, *atlas.Response, error)
+}
+
```

Comment:
[q] Why aren’t we adding this into the store package?

Reply:
this is not store related bot `oauth` related it could be added in the `oauth` package but then only this command should be using this interface, it's a good practice to define the interface when is needed and not where is implemented